# 코로나 전후 상권 변화 분석 - 데이터 전처리 (1)
## 1. 패키지 참조

In [1]:
import numpy as np
from pandas import read_csv, concat, merge, DataFrame
from matplotlib import pyplot as plt
import seaborn as sb

## 2. 데이터 구축
### 1) 데이터 열기

In [2]:
df2018 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권-추정매출)_2018.csv', encoding='euc-kr');
df2019 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권-추정매출)_2019.csv', encoding='euc-kr');
df2020 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권-추정매출)_2020.csv', encoding='euc-kr');
df2021 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권-추정매출)_2021.csv', encoding='euc-kr');
df상권영역 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권영역).csv', encoding='euc-kr');
df건축물 = read_csv('/Users/dayoonz/Desktop/data/서울특별시 건축물대장 법정동 코드정보.csv', encoding='euc-kr');
df직장인구2018 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-직장인구)-2018.csv', encoding='euc-kr')
df직장인구2019 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-직장인구)-2019.csv', encoding='euc-kr')
df직장인구2020 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-직장인구)-2020.csv', encoding='euc-kr')
df직장인구2021 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-직장인구)-2021.csv', encoding='euc-kr')
df생활인구2018 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-생활인구)-2018.csv', encoding='euc-kr')
df생활인구2019 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-생활인구)-2019.csv', encoding='euc-kr')
df생활인구2020 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-생활인구)-2020.csv', encoding='euc-kr')
df생활인구2021 = read_csv('/Users/dayoonz/Desktop/data/서울시 우리마을가게 상권분석서비스(상권배후지-생활인구)-2021.csv', encoding='euc-kr')

### 2) 상권영역에 데이터 구축하기
#### 상권영역 데이터 확인
- 상권영역 데이터는 서울시에만 해당하는 데이터이다

In [4]:
df상권영역.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1496 entries, 0 to 1495
Data columns (total 10 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   기준_년월_코드    1496 non-null   int64  
 1   상권_구분_코드    1496 non-null   object 
 2   상권_구분_코드_명  1496 non-null   object 
 3   상권_코드       1496 non-null   int64  
 4   상권_코드_명     1496 non-null   object 
 5   엑스좌표_값      1496 non-null   int64  
 6   와이좌표_값      1496 non-null   int64  
 7   시군구_코드      1496 non-null   int64  
 8   행정동_코드      1496 non-null   int64  
 9   형태정보        0 non-null      float64
dtypes: float64(1), int64(6), object(3)
memory usage: 117.0+ KB


#### 법정동 코드정보 확인
- 법정동 코드정보는 전국에 대한 데이터이다

In [5]:
df건축물['시군구코드'].value_counts()

시군구코드
47780    970
47740    921
47800    875
47810    835
43710    818
        ... 
27145      1
28265      1
25000      1
45118      1
22235      1
Name: count, Length: 500, dtype: int64

> `47780`이라는 코드를 갖는 지역이 `970`곳이 됨을 확인할 수 있다. 즉, `코드값의 중복`이 있음을 확인.

#### 시군구명, 시군구코드를 추출하여 중복데이터 제거

In [6]:
df시군구 = df건축물.filter(['시군구명', '시군구코드'])
df시군구 = df시군구.drop_duplicates()
df시군구.dropna(inplace=True)
df시군구

시군구코드      시군구명
0      42750       영월군
14     30200       유성구
23     41220       평택시
30     41465   용인시 수지구
32     41461   용인시 처인구
...      ...       ...
82870  44170       대천시
83192  44190       온양시
86860  22145  동구 안심출장소
88397  42105     동해출장소
92806  22235  북구 칠곡출장소

[476 rows x 2 columns]

#### 시군구명과 시군구코드를 상권영역에 결합

In [7]:
df상권영역2 = merge(df상권영역, 
                df시군구,
                left_on = '시군구_코드', 
                right_on = '시군구코드',
                how='left')

df상권영역2

기준_년월_코드 상권_구분_코드 상권_구분_코드_명    상권_코드          상권_코드_명  엑스좌표_값  와이좌표_값  \
0       201810        R       전통시장  1001453            낙성대시장  196121  442084   
1       201810        R       전통시장  1001454         봉천제일종합시장  195147  442413   
2       201810        R       전통시장  1001474             도곡시장  204551  444227   
3       201810        R       전통시장  1001475           강남개포시장  206065  443310   
4       201810        R       전통시장  1001412           화곡본동시장  186203  449328   
...        ...      ...        ...      ...              ...     ...     ...   
1491    201810        D       발달상권  1001044            양재역_3  202755  443008   
1492    201810        D       발달상권  1001030            양재역_2  203216  442691   
1493    201810        D       발달상권  1001031     서울 관악구 신림역_1  193706  442615   
1494    201810        D       발달상권  1001032     서울 관악구 신림역_2  193880  442602   
1495    201810        D       발달상권  1001033  서울 서초구 남부터미널역_1  201661  442764   

      시군구_코드    행정동_코드  형태정보  시군구코드 시군구명  
0      11620  11620585   NaN  11620  관악구  
1      11620  11620595   NaN  11620  관악구  
2      11680  11680650   NaN  11680  강남구  
3      11680  11680670   NaN  11680  강남구  
4      11500  11500590   NaN  11500  강서구  
...      ...       ...   ...    ...  ...  
1491   11650  11650520   NaN  11650  서초구  
1492   11650  11650651   NaN  11650  서초구  
1493   11620  11620645   NaN  11620  관악구  
1494   11620  11620645   NaN  11620  관악구  
1495   11650  11650510   NaN  11650  서초구  

[1496 rows x 12 columns]

#### `구`에 대한 정보 확인
- 얼마나 많은 구가 있는가?

In [8]:
df상권영역2['시군구명'].value_counts()

시군구명
강남구     115
관악구      88
동대문구     79
영등포구     74
마포구      71
서초구      70
성북구      69
강서구      69
동작구      65
양천구      63
종로구      63
은평구      61
강동구      60
광진구      60
구로구      59
용산구      58
중구       55
강북구      52
성동구      48
중랑구      48
송파구      45
금천구      41
도봉구      33
서대문구     32
노원구      18
Name: count, dtype: int64

#### 파생변수 생성
- `행정구`는 값의 종류가 너무 많아서 더미변수화 하기가 어려울 것으로 보고 `도심과 비도심으로 구분`하는 파생변수를 생성
> (출처: 서울2030플렌)
> 
> 서울의 3대도심: 행정도심, 금융도심, 경제도심
> 
> 행정도심: 종로구 & 중구, 금융도심: 영등포구 (여의도 포함), 경제도심: 서초구 & 강남구

In [10]:
도심 = ['종로구', '중구', '영등포구', '서초구', '강남구']
df상권영역2['도심'] = np.isin(df상권영역2['시군구명'], 도심)
df상권영역2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1496 entries, 0 to 1495
Data columns (total 13 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   기준_년월_코드    1496 non-null   int64  
 1   상권_구분_코드    1496 non-null   object 
 2   상권_구분_코드_명  1496 non-null   object 
 3   상권_코드       1496 non-null   int64  
 4   상권_코드_명     1496 non-null   object 
 5   엑스좌표_값      1496 non-null   int64  
 6   와이좌표_값      1496 non-null   int64  
 7   시군구_코드      1496 non-null   int64  
 8   행정동_코드      1496 non-null   int64  
 9   형태정보        0 non-null      float64
 10  시군구코드       1496 non-null   int64  
 11  시군구명        1496 non-null   object 
 12  도심          1496 non-null   bool   
dtypes: bool(1), float64(1), int64(7), object(4)
memory usage: 141.8+ KB


#### 상권영역 데이터 저장
- `df.to_excel(filename='')`

In [11]:
# df상권영역2.to_excel('상권영역.xlsx', index=False)

### 3) 코로나 전후 매출 데이터
#### 코로나 이전 매출 데이터
- 단순 데이터 결합

In [19]:
코로나이전df = concat([df2018, df2019])
코로나이전df.reset_index(drop=True, inplace=True)
코로나이전df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 298244 entries, 0 to 298243
Data columns (total 80 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   기준_년_코드          298244 non-null  int64 
 1   기준_분기_코드         298244 non-null  int64 
 2   상권_구분_코드         298244 non-null  object
 3   상권_구분_코드_명       298244 non-null  object
 4   상권_코드            298244 non-null  int64 
 5   상권_코드_명          298244 non-null  object
 6   서비스_업종_코드        298244 non-null  object
 7   서비스_업종_코드_명      298244 non-null  object
 8   분기당_매출_금액        298244 non-null  int64 
 9   분기당_매출_건수        298244 non-null  int64 
 10  주중_매출_비율         298244 non-null  int64 
 11  주말_매출_비율         298244 non-null  int64 
 12  월요일_매출_비율        298244 non-null  int64 
 13  화요일_매출_비율        298244 non-null  int64 
 14  수요일_매출_비율        298244 non-null  int64 
 15  목요일_매출_비율        298244 non-null  int64 
 16  금요일_매출_비율        298244 non-null  int64 
 17  토요일_매출_비율 

- 골목상권 데이터만 추출

In [21]:
코로나이전df2 = 코로나이전df.query("상권_구분_코드_명 == '골목상권'")
코로나이전df2.head()

기준_년_코드  기준_분기_코드 상권_구분_코드 상권_구분_코드_명    상권_코드   상권_코드_명 서비스_업종_코드  \
2     2018         1        A       골목상권  1000289   인촌로17가길  CS300021   
3     2018         1        A       골목상권  1000515  남부순환로70길  CS200030   
4     2018         1        A       골목상권  1000902    논현로63길  CS200029   
5     2018         1        A       골목상권  1000407    연서로33길  CS200007   
6     2018         1        A       골목상권  1000407    연서로33길  CS200008   

  서비스_업종_코드_명  분기당_매출_금액  분기당_매출_건수  ...  시간대_건수~24_매출_건수  남성_매출_건수  여성_매출_건수  \
2          문구    6044732         23  ...                0        23         0   
3       피부관리실   11863957         40  ...                0         0        36   
4         네일숍    1766586         19  ...                3         0        15   
5        치과의원  140020113       2150  ...                0      1033      1117   
6         한의원  145038784       6360  ...               20      2712      3648   

   연령대_10_매출_건수  연령대_20_매출_건수  연령대_30_매출_건수  연령대_40_매출_건수  연령대_50_매출_건수  \
2             0             0             0             8             7   
3             0             0             0            13            23   
4             0             0            11             0             0   
5            32           280           324           402           458   
6            10           391           896          1326          1930   

   연령대_60_이상_매출_건수  점포수  
2                8    4  
3                0    1  
4                5    2  
5              654    1  
6             1807    1  

[5 rows x 80 columns]

- 코로나 이전과 이후의 조사범위를 동일하게 설정
    - 코로나19 발생시점인 2020년 1월을 기준으로 2018년 1, 2, 3, 4분기, 2019년 1, 2, 3, 4분기 데이터와 2020년 1, 2, 3, 4분기, 2021년 1, 2분기 데이터만 수집된 상태
    - 그러므로 코로나 이전 데이터에서 2018년 1, 2분기를 `분석 범위에서 제외`해야 코로나 전후에 대한 `데이터 분포가 균형`을 이루게 된다

In [24]:
제외할index = 코로나이전df2.query("(기준_년_코드==2018 and (기준_분기_코드==1 or 기준_분기_코드==2))").index
제외할index

코로나이전df3 = 코로나이전df2.drop(제외할index)
코로나이전df3.reset_index(drop=True, inplace=True)



코로나이전df3.reset_index(drop=True, inplace=True)
# 코로나이전df3.to_excel('코로나이전_매출데이터.xlsx', index=False)
코로나이전df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139608 entries, 0 to 139607
Data columns (total 80 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   기준_년_코드          139608 non-null  int64 
 1   기준_분기_코드         139608 non-null  int64 
 2   상권_구분_코드         139608 non-null  object
 3   상권_구분_코드_명       139608 non-null  object
 4   상권_코드            139608 non-null  int64 
 5   상권_코드_명          139608 non-null  object
 6   서비스_업종_코드        139608 non-null  object
 7   서비스_업종_코드_명      139608 non-null  object
 8   분기당_매출_금액        139608 non-null  int64 
 9   분기당_매출_건수        139608 non-null  int64 
 10  주중_매출_비율         139608 non-null  int64 
 11  주말_매출_비율         139608 non-null  int64 
 12  월요일_매출_비율        139608 non-null  int64 
 13  화요일_매출_비율        139608 non-null  int64 
 14  수요일_매출_비율        139608 non-null  int64 
 15  목요일_매출_비율        139608 non-null  int64 
 16  금요일_매출_비율        139608 non-null  int64 
 17  토요일_매출_비율 

#### 코로나 이후 매출 데이터

In [26]:
코로나이후df = concat([df2020, df2021])
코로나이후df.reset_index(drop=True, inplace=True)

코로나이후df2 = 코로나이후df.query('상권_구분_코드_명 == "골목상권"')
# 코로나이후df2.to_excel('코로나이후_매출데이터.xlsx', index=False)
코로나이후df2.info()

<class 'pandas.core.frame.DataFrame'>
Index: 126320 entries, 2 to 197058
Data columns (total 80 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   기준_년_코드          126320 non-null  int64  
 1   기준_분기_코드         126320 non-null  int64  
 2   상권_구분_코드         126320 non-null  object 
 3   상권_구분_코드_명       126320 non-null  object 
 4   상권_코드            126320 non-null  int64  
 5   상권_코드_명          126320 non-null  object 
 6   서비스_업종_코드        126320 non-null  object 
 7   서비스_업종_코드_명      126320 non-null  object 
 8   분기당_매출_금액        126320 non-null  float64
 9   분기당_매출_건수        126320 non-null  int64  
 10  주중_매출_비율         126320 non-null  int64  
 11  주말_매출_비율         126320 non-null  int64  
 12  월요일_매출_비율        126320 non-null  int64  
 13  화요일_매출_비율        126320 non-null  int64  
 14  수요일_매출_비율        126320 non-null  int64  
 15  목요일_매출_비율        126320 non-null  int64  
 16  금요일_매출_비율        126320 non-null  int64  
 

#### 코로나 전, 후 데이터 결합
- 코로나 이전 데이터에 명목형 변수 추가 (코로나 전인지 후인지)
- 코로나 전이면 0, 후면 1

In [28]:
코로나이전df3['Covid19'] = 0
코로나이전df3.shape
코로나이전df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139608 entries, 0 to 139607
Data columns (total 81 columns):
 #   Column           Non-Null Count   Dtype 
---  ------           --------------   ----- 
 0   기준_년_코드          139608 non-null  int64 
 1   기준_분기_코드         139608 non-null  int64 
 2   상권_구분_코드         139608 non-null  object
 3   상권_구분_코드_명       139608 non-null  object
 4   상권_코드            139608 non-null  int64 
 5   상권_코드_명          139608 non-null  object
 6   서비스_업종_코드        139608 non-null  object
 7   서비스_업종_코드_명      139608 non-null  object
 8   분기당_매출_금액        139608 non-null  int64 
 9   분기당_매출_건수        139608 non-null  int64 
 10  주중_매출_비율         139608 non-null  int64 
 11  주말_매출_비율         139608 non-null  int64 
 12  월요일_매출_비율        139608 non-null  int64 
 13  화요일_매출_비율        139608 non-null  int64 
 14  수요일_매출_비율        139608 non-null  int64 
 15  목요일_매출_비율        139608 non-null  int64 
 16  금요일_매출_비율        139608 non-null  int64 
 17  토요일_매출_비율 

- 코로나 이후 데이터에 명목형 변수 추가

In [32]:
코로나이후df2['Covid19'] = 1

# 코로나이후df2.shape
# 코로나이후df2['Covid19'].value_counts()
# 코로나이후df2.info()

/var/folders/md/0jpjv9rd2qg156cjt56sczy80000gn/T/ipykernel_1506/2007934780.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  코로나이후df2['Covid19'] = 1


- 코로나 전후 매출데이터 병합

In [34]:
매출데이터_최종 = concat([코로나이전df3, 코로나이후df2])
매출데이터_최종.reset_index(drop=True, inplace=True)
# print(매출데이터_최종.shape)
# print(매출데이터_최종.info())
# print(매출데이터_최종.head())

- 분석에 사용할 요인만 추출 및 파생변수 생성

In [35]:
분석데이터 = 매출데이터_최종.filter(['기준_년_코드', '기준_분기_코드', 
                              'Covid19', '상권_코드', 
                              '분기당_매출_금액', '주중_매출_금액', 
                              '주말_매출_금액', '시간대_00~06_매출_금액', 
                              '시간대_06~11_매출_금액', '시간대_11~14_매출_금액', 
                              '시간대_14~17_매출_금액', '시간대_17~21_매출_금액', 
                              '시간대_21~24_매출_금액', '남성_매출_금액', 
                              '여성_매출_금액', '연령대_10_매출_금액', 
                              '연령대_20_매출_금액', '연령대_30_매출_금액', 
                              '연령대_40_매출_금액', '연령대_50_매출_금액', 
                              '연령대_60_이상_매출_금액', '점포수'], 
                              axis=1)
분석데이터['일과시간_금액'] = 분석데이터['시간대_06~11_매출_금액'] + 분석데이터['시간대_11~14_매출_금액'] + 분석데이터['시간대_14~17_매출_금액']
분석데이터['일과이후_금액'] = 분석데이터['시간대_17~21_매출_금액'] + 분석데이터['시간대_21~24_매출_금액'] + 분석데이터['시간대_00~06_매출_금액']
분석데이터['청년금액'] = (분석데이터['연령대_10_매출_금액'] + 분석데이터['연령대_20_매출_금액'] + 분석데이터['연령대_30_매출_금액'])
분석데이터['중년금액'] = (분석데이터['연령대_40_매출_금액'] + 분석데이터['연령대_50_매출_금액'])
분석데이터['장년금액'] = (분석데이터['연령대_60_이상_매출_금액'])

분석데이터.drop(['시간대_00~06_매출_금액', '시간대_06~11_매출_금액', '시간대_11~14_매출_금액', 
              '시간대_14~17_매출_금액', '시간대_17~21_매출_금액', '시간대_21~24_매출_금액', 
              '연령대_10_매출_금액', '연령대_20_매출_금액', '연령대_30_매출_금액', 
              '연령대_40_매출_금액', '연령대_50_매출_금액', '연령대_60_이상_매출_금액'], 
              axis=1, inplace=True)

# 분석데이터.to_excel('분석데이터.xlsx', index=False)

# print(분석데이터.shape)
# print(분석데이터.info())
# print(분석데이터.head())

### 4) 매출데이터에 상권영역 정보를 결합
#### 결합할 변수만 추출

In [38]:
tmp = df상권영역2.filter(['상권_코드','상권_코드_명','엑스좌표_값','와이좌표_값',
                        '시군구_코드','행정동_코드','시군구명','시군구코드','도심'])
tmp =  tmp.drop_duplicates()
tmp.dropna(inplace=True)
tmp.shape

(1496, 9)

#### 매출데이터에 상권영역 정보 결합

In [39]:
상권매출df = merge(분석데이터, 
                tmp, 
                left_on = '상권_코드',
                right_on = '상권_코드',
                how = 'inner')
상권매출df.shape
상권매출df.head()

기준_년_코드  기준_분기_코드    상권_코드     분기당_매출_금액      주중_매출_금액     주말_매출_금액  \
0     2018         3  1000094  1.824475e+08  1.816677e+08     779766.0   
1     2018         3  1000094  4.561081e+06  3.428057e+06    1133024.0   
2     2018         3  1000094  8.629311e+07  7.820822e+07    8084889.0   
3     2018         3  1000094  9.394161e+07  6.050822e+07   33433384.0   
4     2018         3  1000094  1.692325e+09  1.354745e+09  337580548.0   

      남성_매출_금액     여성_매출_금액  점포수  Covid19  ...         중년금액       장년금액  \
0   81108758.0   83644304.0    5        0  ...  153850431.0          0   
1          0.0    4561081.0    2        0  ...    4291743.0          0   
2   38252844.0   48040265.0    1        0  ...   69039046.0          0   
3   45574085.0   36140265.0    1        0  ...   34418855.0    4464070   
4  489138378.0  195886445.0    1        0  ...  367308046.0  178434388   

   상권_코드_명  엑스좌표_값  와이좌표_값 시군구_코드    행정동_코드  시군구코드  시군구명     도심  
0   마장로39길  203915  452058  11200  11200540  11200   성동구  False  
1   마장로39길  203915  452058  11200  11200540  11200   성동구  False  
2   마장로39길  203915  452058  11200  11200540  11200   성동구  False  
3   마장로39길  203915  452058  11200  11200540  11200   성동구  False  
4   마장로39길  203915  452058  11200  11200540  11200   성동구  False  

[5 rows x 23 columns]

### 5) 인구 데이터
#### 코로나 이전 배후지 직장 인구 데이터
- 데이터 확인

In [40]:
print(df직장인구2018.shape)
print(df직장인구2018.info())
df직장인구2018.head()

(4040, 27)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4040 entries, 0 to 4039
Data columns (total 27 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   기준_년_코드              4040 non-null   int64 
 1   기준_분기_코드             4040 non-null   int64 
 2   상권_구분_코드             4040 non-null   object
 3   상권_구분_코드_명           4040 non-null   object
 4   상권_코드                4040 non-null   int64 
 5   상권_코드_명              4040 non-null   object
 6   총_직장_인구_수            4040 non-null   int64 
 7   남성_직장_인구_수           4040 non-null   int64 
 8   여성_직장_인구_수           4040 non-null   int64 
 9   연령대_10_직장_인구_수       4040 non-null   int64 
 10  연령대_20_직장_인구_수       4040 non-null   int64 
 11  연령대_30_직장_인구_수       4040 non-null   int64 
 12  연령대_40_직장_인구_수       4040 non-null   int64 
 13  연령대_50_직장_인구_수       4040 non-null   int64 
 14  연령대_60_이상_직장_인구_수    4040 non-null   int64 
 15  남성연령대_10_직장_인구_수     4040 non-null   int64 


기준_년_코드  기준_분기_코드 상권_구분_코드 상권_구분_코드_명    상권_코드    상권_코드_명  총_직장_인구_수  \
0     2018         4        A       골목상권  1000159     자양로15길       6423   
1     2018         4        A       골목상권  1000934      일원로9길       8245   
2     2018         4        A       골목상권  1000819     신림로48길        825   
3     2018         4        A       골목상권  1000797  남부순환로161길       2836   
4     2018         4        A       골목상권  1000692     대림로22길       2779   

   남성_직장_인구_수  여성_직장_인구_수  연령대_10_직장_인구_수  ...  남성연령대_30_직장_인구_수  \
0        4133        2290              20  ...               702   
1        2229        6016               2  ...               727   
2         344         481               0  ...                72   
3        1575        1261               2  ...               364   
4        1349        1430               9  ...               218   

   남성연령대_40_직장_인구_수  남성연령대_50_직장_인구_수  남성연령대_60_이상_직장_인구_수  여성연령대_10_직장_인구_수  \
0              1241              1269                  601                10   
1               537               385                  229                 2   
2                73                83                   78                 0   
3               437               354                  309                 0   
4               314               330                  384                 6   

   여성연령대_20_직장_인구_수  여성연령대_30_직장_인구_수  여성연령대_40_직장_인구_수  여성연령대_50_직장_인구_수  \
0               490               489               519               548   
1              1997              2093              1153               601   
2                96                99               109               115   
3               197               311               300               308   
4               294               271               329               326   

   여성연령대_60_이상_직장_인구_수  
0                  234  
1                  170  
2                   62  
3                  145  
4                  204  

[5 rows x 27 columns]

In [41]:
코로나이전_직장인구_df = concat([df직장인구2018, df직장인구2019])
코로나이전_직장인구_df.reset_index(drop=True, inplace=True)
drop_index = 코로나이전_직장인구_df.query("기준_년_코드==2018 and (기준_분기_코드==1 or 기준_분기_코드==2)").index
코로나이전_직장인구_df.drop(drop_index, inplace=True)
코로나이전_직장인구_df = 코로나이전_직장인구_df.query('상권_구분_코드_명 == "골목상권"')
코로나이전_직장인구_df.reset_index(drop=True, inplace=True)
코로나이전_직장인구df = 코로나이전_직장인구_df.filter(['기준_년_코드', '기준_분기_코드', '상권_코드', '총_직장_인구_수'])
# 코로나이전_직장인구df.head()

코로나이전_직장인구df.groupby(['기준_년_코드', '기준_분기_코드']).count()

상권_코드  총_직장_인구_수
기준_년_코드 기준_분기_코드                  
2018    3          1010       1010
        4          1010       1010
2019    1          1010       1010
        2          1010       1010
        3          1010       1010
        4          1010       1010

#### 코로나 이후 배후지 직장 인구 데이터

In [45]:
코로나이후_직장인구_df = concat([df직장인구2020, df직장인구2021])
코로나이전_직장인구_df.reset_index(drop=True, inplace=True)
코로나이후_직장인구_df = 코로나이후_직장인구_df.query('상권_구분_코드_명 == "골목상권"')
코로나이후_직장인구_df.reset_index(drop=True, inplace=True)
코로나이후_직장인구df = 코로나이후_직장인구_df.filter(['기준_년_코드', '기준_분기_코드', '상권_코드', '총_직장_인구_수'])
# 코로나이후_직장인구df.head()

코로나이후_직장인구df.groupby(['기준_년_코드', '기준_분기_코드']).count()

상권_코드  총_직장_인구_수
기준_년_코드 기준_분기_코드                  
2020    1          1010       1010
        2          1010       1010
        3          1010       1010
        4          1010       1010
2021    1          1010       1010
        2          1010       1010

#### 코로나 이전 배후지 생활 인구 데이터
- 데이터 확인

In [48]:
df생활인구2018.info()
df생활인구2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4040 entries, 0 to 4039
Columns: 532 entries, 기준_년_코드 to 여성연령대_60_이상_일요일시간대_6_생활인구_수
dtypes: int64(529), object(3)
memory usage: 16.4+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4040 entries, 0 to 4039
Columns: 532 entries, 기준_년_코드 to 여성연령대_60_이상_일요일시간대_6_생활인구_수
dtypes: int64(529), object(3)
memory usage: 16.4+ MB


- 데이터 병합 및 추출

In [49]:
코로나이전_생활인구_df = concat([df생활인구2018, df생활인구2019])
코로나이전_생활인구_df.reset_index(drop=True, inplace=True)
drop_index = 코로나이전_생활인구_df.query("(기준_년_코드==2018 and (기준_분기_코드==1 or 기준_분기_코드==2))").index
코로나이전_생활인구_df.drop(drop_index, inplace=True)
코로나이전_생활인구_df = 코로나이전_생활인구_df.query('상권_구분_코드_명 == "골목상권"')
코로나이전_생활인구_df.reset_index(drop=True, inplace=True)
코로나이전_생활인구df = 코로나이전_생활인구_df.filter(['기준_년_코드', '기준_분기_코드', '상권_코드', '총_생활인구_수'])
코로나이전_생활인구df.head()

기준_년_코드  기준_분기_코드    상권_코드  총_생활인구_수
0     2018         4  1000005     53242
1     2018         4  1000107    752834
2     2018         4  1000403   2007766
3     2018         4  1000094    775804
4     2018         4  1000650   1349770

#### 코로나 이후 배후지 생활 인구 데이터
- 데이터 병합 및 추출

In [50]:
코로나이후_생활인구_df = concat([df생활인구2020, df생활인구2021])
코로나이후_생활인구_df.reset_index(drop=True, inplace=True)
코로나이후_생활인구_df = 코로나이후_생활인구_df.query('상권_구분_코드_명 == "골목상권"')
코로나이후_생활인구_df.reset_index(drop=True, inplace=True)
코로나이후_생활인구df = 코로나이후_생활인구_df.filter(['기준_년_코드', '기준_분기_코드', '상권_코드', '총_생활인구_수'])
코로나이후_생활인구df.head()

기준_년_코드  기준_분기_코드    상권_코드  총_생활인구_수
0     2020         4  1000117   4561311
1     2020         4  1000524    660691
2     2020         4  1000101   5400680
3     2020         4  1000777   4587594
4     2020         4  1000530   3339989

#### 인구 데이터 결합
- 코로나 전후 직장인구 데이터 결합

In [51]:
직장인구 = concat([코로나이전_직장인구df, 코로나이후_직장인구df])
print(직장인구.shape)
print(직장인구.info())
직장인구.head()

(12120, 4)
<class 'pandas.core.frame.DataFrame'>
Index: 12120 entries, 0 to 6059
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   기준_년_코드    12120 non-null  int64
 1   기준_분기_코드   12120 non-null  int64
 2   상권_코드      12120 non-null  int64
 3   총_직장_인구_수  12120 non-null  int64
dtypes: int64(4)
memory usage: 473.4 KB
None


기준_년_코드  기준_분기_코드    상권_코드  총_직장_인구_수
0     2018         4  1000159       6423
1     2018         4  1000934       8245
2     2018         4  1000819        825
3     2018         4  1000797       2836
4     2018         4  1000692       2779

In [52]:
직장인구.groupby(['기준_년_코드', '기준_분기_코드']).count()

상권_코드  총_직장_인구_수
기준_년_코드 기준_분기_코드                  
2018    3          1010       1010
        4          1010       1010
2019    1          1010       1010
        2          1010       1010
        3          1010       1010
        4          1010       1010
2020    1          1010       1010
        2          1010       1010
        3          1010       1010
        4          1010       1010
2021    1          1010       1010
        2          1010       1010

- 코로나 전후 생활인구 데이터 결합

In [53]:
생활인구 = concat([코로나이전_생활인구df, 코로나이후_생활인구df])
print(생활인구.shape)
print(생활인구.info())
생활인구.head()

(12120, 4)
<class 'pandas.core.frame.DataFrame'>
Index: 12120 entries, 0 to 6059
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   기준_년_코드   12120 non-null  int64
 1   기준_분기_코드  12120 non-null  int64
 2   상권_코드     12120 non-null  int64
 3   총_생활인구_수  12120 non-null  int64
dtypes: int64(4)
memory usage: 473.4 KB
None


기준_년_코드  기준_분기_코드    상권_코드  총_생활인구_수
0     2018         4  1000005     53242
1     2018         4  1000107    752834
2     2018         4  1000403   2007766
3     2018         4  1000094    775804
4     2018         4  1000650   1349770

- 직장인구, 생활인구 데이터 병합

In [54]:
인구데이터 = merge(직장인구, 
                생활인구, 
                left_on=['기준_년_코드','기준_분기_코드','상권_코드'], 
                right_on=['기준_년_코드','기준_분기_코드','상권_코드'], 
                how='outer')

print(인구데이터.shape)
print(인구데이터.info())
# 인구데이터.to_excel('인구데이터.xlsx', index=False)
인구데이터.head()

(12120, 5)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12120 entries, 0 to 12119
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   기준_년_코드    12120 non-null  int64
 1   기준_분기_코드   12120 non-null  int64
 2   상권_코드      12120 non-null  int64
 3   총_직장_인구_수  12120 non-null  int64
 4   총_생활인구_수   12120 non-null  int64
dtypes: int64(5)
memory usage: 473.6 KB
None


기준_년_코드  기준_분기_코드    상권_코드  총_직장_인구_수  총_생활인구_수
0     2018         4  1000159       6423   2239424
1     2018         4  1000934       8245   1140300
2     2018         4  1000819        825    835644
3     2018         4  1000797       2836   1240350
4     2018         4  1000692       2779   1631737

### 6) 최종 데이터 프레임 구성
#### (매출데이터 + 상권데이터) + 인구데이터

In [55]:
전처리완료df = merge(상권매출df, 
                  인구데이터, 
                  left_on=['기준_년_코드','기준_분기_코드','상권_코드'], 
                  right_on=['기준_년_코드','기준_분기_코드','상권_코드'],
                  how='outer')
print(전처리완료df.shape)
print(전처리완료df.info())
전처리완료df.head()

(265937, 25)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 265937 entries, 0 to 265936
Data columns (total 25 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   기준_년_코드    265937 non-null  int64  
 1   기준_분기_코드   265937 non-null  int64  
 2   상권_코드      265937 non-null  int64  
 3   분기당_매출_금액  265928 non-null  float64
 4   주중_매출_금액   265928 non-null  float64
 5   주말_매출_금액   265928 non-null  float64
 6   남성_매출_금액   265928 non-null  float64
 7   여성_매출_금액   265928 non-null  float64
 8   점포수        265928 non-null  float64
 9   Covid19    265928 non-null  float64
 10  일과시간_금액    265928 non-null  float64
 11  일과이후_금액    265928 non-null  float64
 12  청년금액       265928 non-null  float64
 13  중년금액       265928 non-null  float64
 14  장년금액       265928 non-null  float64
 15  상권_코드_명    265928 non-null  object 
 16  엑스좌표_값     265928 non-null  float64
 17  와이좌표_값     265928 non-null  float64
 18  시군구_코드     265928 non-null  float64
 19  행정동_코드    

기준_년_코드  기준_분기_코드    상권_코드     분기당_매출_금액      주중_매출_금액     주말_매출_금액  \
0     2018         3  1000094  1.824475e+08  1.816677e+08     779766.0   
1     2018         3  1000094  4.561081e+06  3.428057e+06    1133024.0   
2     2018         3  1000094  8.629311e+07  7.820822e+07    8084889.0   
3     2018         3  1000094  9.394161e+07  6.050822e+07   33433384.0   
4     2018         3  1000094  1.692325e+09  1.354745e+09  337580548.0   

      남성_매출_금액     여성_매출_금액  점포수  Covid19  ...  상권_코드_명    엑스좌표_값    와이좌표_값  \
0   81108758.0   83644304.0  5.0      0.0  ...   마장로39길  203915.0  452058.0   
1          0.0    4561081.0  2.0      0.0  ...   마장로39길  203915.0  452058.0   
2   38252844.0   48040265.0  1.0      0.0  ...   마장로39길  203915.0  452058.0   
3   45574085.0   36140265.0  1.0      0.0  ...   마장로39길  203915.0  452058.0   
4  489138378.0  195886445.0  1.0      0.0  ...   마장로39길  203915.0  452058.0   

    시군구_코드      행정동_코드    시군구코드  시군구명     도심  총_직장_인구_수  총_생활인구_수  
0  11200.0  11200540.0  11200.0   성동구  False       3118   1272175  
1  11200.0  11200540.0  11200.0   성동구  False       3118   1272175  
2  11200.0  11200540.0  11200.0   성동구  False       3118   1272175  
3  11200.0  11200540.0  11200.0   성동구  False       3118   1272175  
4  11200.0  11200540.0  11200.0   성동구  False       3118   1272175  

[5 rows x 25 columns]

#### 결측치 확인

In [56]:
전처리완료df.isna().sum()

기준_년_코드      0
기준_분기_코드     0
상권_코드        0
분기당_매출_금액    9
주중_매출_금액     9
주말_매출_금액     9
남성_매출_금액     9
여성_매출_금액     9
점포수          9
Covid19      9
일과시간_금액      9
일과이후_금액      9
청년금액         9
중년금액         9
장년금액         9
상권_코드_명      9
엑스좌표_값       9
와이좌표_값       9
시군구_코드       9
행정동_코드       9
시군구코드        9
시군구명         9
도심           9
총_직장_인구_수    0
총_생활인구_수     0
dtype: int64

> 미미한 수이므로 제거하기로 결정

#### 결측치 제거

In [57]:
전처리완료df.dropna(inplace=True)
# 전처리완료df.to_excel("분석용데이터.xlsx", index=False)
전처리완료df.isna().sum()

기준_년_코드      0
기준_분기_코드     0
상권_코드        0
분기당_매출_금액    0
주중_매출_금액     0
주말_매출_금액     0
남성_매출_금액     0
여성_매출_금액     0
점포수          0
Covid19      0
일과시간_금액      0
일과이후_금액      0
청년금액         0
중년금액         0
장년금액         0
상권_코드_명      0
엑스좌표_값       0
와이좌표_값       0
시군구_코드       0
행정동_코드       0
시군구코드        0
시군구명         0
도심           0
총_직장_인구_수    0
총_생활인구_수     0
dtype: int64